src/foam/dao/AbstractJDAO.java

```diff
@@ -79,20 +79,26 @@ protected void loadJournal()
         // remove first two characters and last character
         line = line.trim().substring(2, line.trim().length() - 1);
 
-        FObject object = parser.parseString(line);
-        if ( object == null ) {
-          System.err.println(getParsingErrorMessage(line) + ", source: " + line);
-          break;
-        }
-
+        FObject object;
         switch ( operation ) {
           case 'p':
-            getDelegate().put(object);
+            object = parser.parseString(line);
+            if ( object != null ) {
+              getDelegate().put(object);
+            }
             break;
           case 'r':
-            getDelegate().remove(object);
+            Object id = parser.parseStringForId(line);
```

Comment:
There's a reason I removed the parseStringForId line and used parseObject in both cases.

parseStringForId fails for multipart ID objects.

Reply:
Okay then how do you propose we solve this. The current remove lines that are printed out look like this: ```r({"id":1})``` and parseString does not parse this either

Reply:
I changed the remove lines to output the full object as well when I made this change.  I propose leaving it as is.

Reply:
https://github.com/foam-framework/foam2/commit/a501a9c7b6544154aa9aadae7aa93578c43a495a

Reply:
Actually I see that you modified the remove_ function to print out the whole object just like the put_ method. But as it stands the old remove lines are broken.

Reply:
Do we have existing data in the old format we need to support?

I guess we could try to parse it both ways.  If the current method fails to parse, then try to parse it as just an id.

Reply:
@adamvy Okay leaving it as is but instead of stopping on parse failure Ill change it to just continue.